### PR TITLE
Add simulated discovery, services, characteristics

### PR DESCRIPTION
**Pull request by mikolak**
Blocked by #177

### CODE REVIEW

File: android/src/main/java/com/polidea/blemulator/DeviceContainer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    DeviceContainer(String identifier, String name) {
+        this.identifier = identifier;
+        this.name = name;
+    }
+
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    public void setIdentifier(String identifier) {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/blemulator/DeviceContainer.java]: No, deleting

File: android/src/main/java/com/polidea/blemulator/DeviceContainer.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        this.identifier = identifier;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/blemulator/DeviceContainer.java]: No, deleting

File: lib/blemulator/simulation/simulated_peripheral.dart [hunk 3]
```diff
@@ -100,11 +100,7 @@ abstract class SimulatedPeripheral {
   bool isConnected() => _isConnected;
 
   List<SimulatedService> services() {
-    if (!_discoveryDone) {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply lib/blemulator/simulation/simulated_peripheral.dart]: No, deleting

File: android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java [hunk 4]
```diff
@@ -307,4 +345,46 @@ public String getLogLevel() {
...
+        Map<String, List<Characteristic>> characteristics = new HashMap<>();
+        for (Map<String, Object> mappedService : response) {
+            Service service = new Service(
+                    (Integer) mappedService.get(SimulationArgumentName.ID),
+                    deviceIdentifier,
+                    new BluetoothGattService(
+                            UUID.fromString(
+                                    (String) mappedService.get(SimulationArgumentName.UUID)
+                            ),
+                            BluetoothGattService.SERVICE_TYPE_PRIMARY
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java]: We can leave it aside for the time being and revisit this if there's a request to support it.

File: lib/blemulator/internal/simulation_manager_mixins/discovery_mixin.dart [hunk 5]
```diff
@@ -0,0 +1,12 @@
+part of internal;
+
+mixin DiscoveryMixin on SimulationManagerBaseWithErrorChecks {
+  Future<List<SimulatedService>> discovery(String identifier) async {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply lib/blemulator/internal/simulation_manager_mixins/discovery_mixin.dart]: Done

**Comment by bartoszwilk** [comment]
Please create a PR for changes in MBA

**Comment by mikolak** [comment]
> Please create a PR for changes in MBA

https://github.com/Polidea/MultiPlatformBleAdapter/pull/29

**Pull request merged**